src/send-bsd.h

```diff
@@ -23,23 +24,47 @@
 #ifdef ZMAP_SEND_LINUX_H
 #error "Don't include both send-bsd.h and send-linux.h"
 #endif
-#ifdef ZMAP_SEND_MAC_H
-#error "Don't include both send-bsd.h and send-mac.h"
-#endif
-
 
 int send_run_init(UNUSED sock_t sock)
 {
 	// Don't need to do anything on BSD-like variants
 	return EXIT_SUCCESS;
 }
 
-int send_packet(sock_t sock, void *buf, int len, UNUSED uint32_t idx)
+int send_packet(sock_t sock, void *buf, int len, UNUSED uint32_t retry_ct)
 {
-	return write(sock.sock, buf, len);
+	if (zconf.send_ip_pkts) {
+		struct ip *iph = (struct ip *)buf;
+
+#if defined(__APPLE__) || (defined(__FreeBSD__) && __FreeBSD_version < 1100030)
+		// Early BSD's raw IP sockets required IP headers to have len
+		// and off fields in host byte order, as they were being byte
+		// swapped on the way out.  Getting byte order wrong would
+		// result in EINVAL from sendto(2) below.
+		// Most modern BSD systems have fixed this and removed the byte
+		// swapping on raw IP sockets, while some, notably macOS, still
+		// require header fields in host byte order.
+		// See ip(4) for details on byte order requirements of raw IP
+		// sockets.
+		if (retry_ct == 0) {
```

Comment:
I'm not quite sure why this logic would be dependent on this being the first attempt to send this packet (`retry_ct == 0`). Can we add a comment to explain or remove this if it's not needed?

Reply:
Added comment in 0c9316b — it's because the caller passes us the same buffer on retries, still byte-swapped.